src/components/App.js

```diff
@@ -15,14 +36,25 @@ const App = () => {
       .catch(error => {
         setMessage(error.message);
       });
-  });
+		fetchAndSetProducts();
+  }, []);
 
   return (
+  <Router>
     <div className="App">
       <h1>Hello, World!</h1>
       <h2>{ message }</h2>
+      <Route exact path='/products'>
+      	<Products products={products} setProducts={setProducts}/>
+      </Route>
+			{products.map(product => {
+				return <Route exact path ={`/products/${product.id}`}>
```

Comment:
REQUIRED - Instead of mapping over it, use params:
```js
<Route exact path ={`/products/:id`}>
```
then inside the Product component
```js
import {useParams} from 'react-router-dom';
//in your component
const {id} = useParams();
```